src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -256,6 +256,16 @@ public override SingleNamespaceOrTypeDeclaration VisitNamespaceDeclaration(Names
                 diagnostics.Add(ErrorCode.ERR_UnexpectedAliasedName, node.Name.GetLocation());
             }
 
+            if (node.AttributeLists.Count > 0)
+            {
+                diagnostics.Add(ErrorCode.ERR_BadModifiersOnNamespace, node.AttributeLists[0].GetLocation());
+            }
+
+            if (node.Modifiers.Count > 0)
+            {
+                diagnostics.Add(ErrorCode.ERR_BadModifiersOnNamespace, node.Modifiers[0].GetLocation());
+            }
```

Comment:
this moved from LanguageParser to here.  This is how we've dealt with other Namespace errors that were previously syntax errors, that we wanted to move out of hte parser.